### DISCUSSION TITLE
Real time events for the frontend

### DISCUSSION THREAD

**Discussion by aderici** [Budibase/budibase#5455] [Help]
Hi, thanks for the amazing software!

I have a service that provides webhooks to notify me of some events that are time sensitive (like receiving a message or a notification). I'd like to be able to notify my frontend users. What would be the best way to achieve this? 

Best regards,

**Comment by melohagan** [comment]
Hi @aderici 

There is no push notification functionality for budibase apps, however you could use the **Send Email** automation step, or push to a discord / slack channel:

https://docs.budibase.com/docs/automation-actions

**Reply by aderici** [reply]
Thanks for the information, the point was having a websocket type of implementation for the budibase frontend. Can one include custom javascript in the form of an embed? That may solve this issue,

**Reply by melohagan** [reply]
There is another related discussion here: https://github.com/Budibase/budibase/discussions/2608

This was another similar discussion where the user was able to use embedded JS to poll a table for events: https://github.com/Budibase/budibase/discussions/5112#discussioncomment-2456819

He recommended himself to use with caution though!
